### ISSUE TITLE
Why do not include <bos> token in labels while training?

### ISSUE THREAD

**Issue by Ttixan** [chenkaisun/Tetris#1]
I reproduce the Bart generation baseline in your code, but when I check the outcome of generation, I always find that the generation is incomplete. For example, the output is supposed to be "Finished", but I only get "ished".
So I check the datacollator.py in line: 53
```
        if "t5" not in model_name and "gpt2" not in model_name:
            for j, lb in enumerate(labels):
                labels[j] = lb[1:]
```
If I remove these lines of code, I get a expected "Finished". But I wonder why did you decide to clip the `<bos>` in labels?

**Comment by chenkaisun** [comment]
Hi thanks for the interest. I haven't met such incomplete issue before except when training is not long enough, but I'm glad you were able to solve the issue. I clipped bos in label since bos is in the decoder input, and the label token sequence is shifted left during teacher forcing. Thanks.

**Comment by Ttixan** [comment]
I think it may be the problem of different versions of Transformer python library. So did you rewrite some code of Transformer? I see that you include the source code of Transformer in your repository folder.

**Comment by chenkaisun** [comment]
Yes, I use the 4.8.2 version replica of Transformer and have it in my repo. I modified the trainer code a little bit to make it more reproducible.
